### PR TITLE
add weight api

### PR DESCRIPTION
**Pull request by elnatantikal**

### CODE REVIEW

File: android/src/main/kotlin/com/metaflow/flutterhealthfit/FlutterHealthFitPlugin.kt [hunk 1]
```diff
@@ -113,8 +126,19 @@ class FlutterHealthFitPlugin(private val activity: Activity) : MethodCallHandler
...
             if (resultCode == Activity.RESULT_OK) {
-                recordData { success ->
-                    Log.i(TAG, "Record data success: $success!")
+                recordStepsData { success ->
+                    Log.i(TAG, "Record steps data success: $success!")
+
+                    if (success)
+                        deferredResult?.success(true)
+                    else
+                        deferredResult?.error("no record", "Record data operation denied", null)
```
Comment (avi-mastov) [inline android/src/main/kotlin/com/metaflow/flutterhealthfit/FlutterHealthFitPlugin.kt:135]: This is the same error message as for weight. We won't be able to identify where the problem is.

File: android/src/main/kotlin/com/metaflow/flutterhealthfit/FlutterHealthFitPlugin.kt [hunk 2]
```diff
@@ -226,8 +262,38 @@ class FlutterHealthFitPlugin(private val activity: Activity) : MethodCallHandler
         }.start()
     }
 
+    private fun getWeight(startTime: Long, endTime: Long, result: (Float?, Throwable?) -> Unit) {
+        val gsa = GoogleSignIn.getAccountForExtension(activity, getFitnessOptions())
+
+        val request = DataReadRequest.Builder().read(DataType.TYPE_WEIGHT)
+                .setTimeRange(startTime, endTime, TimeUnit.MILLISECONDS)
+                .bucketByTime(1, TimeUnit.DAYS)
+                .setLimit(1)
```
Comment (avi-mastov) [inline android/src/main/kotlin/com/metaflow/flutterhealthfit/FlutterHealthFitPlugin.kt:271]: Does this really fetch the last weight, or the first, or any?

File: example/.flutter-plugins-dependencies [hunk 3]
```diff
@@ -0,0 +1 @@
+{"_info":"// This is a generated file; do not edit or check into version control.","dependencyGraph":[{"name":"flutter_health_fit","dependencies":[]}]}
```
Comment (avi-mastov) [inline example/.flutter-plugins-dependencies:1]: Read the file contents before checking it into version control...

File: example/lib/main.dart [hunk 4]
```diff
@@ -27,14 +28,26 @@ class _MyAppState extends State<MyApp> {
     });
   }
 
+  Future _getLast3DaysWeight() async {
+    final now = DateTime.now().millisecondsSinceEpoch;
+    final last3Days = DateTime.now().add(Duration(days: -4)).millisecondsSinceEpoch;
```
Comment (avi-mastov) [inline example/lib/main.dart:33]: This is not the same `now()` as in the previous line. Better to extract `DateTime.now()` to a final variable.
It is clearer to subtract 4 days than to add -4 days.

File: example/lib/main.dart [hunk 5]
```diff
@@ -27,14 +28,26 @@ class _MyAppState extends State<MyApp> {
...
+    var lastWeight = await FlutterHealthFit().getWeight(last3Days, now);
```
Comment (avi-mastov) [inline example/lib/main.dart:34]:
```suggestion
    final lastWeight = await FlutterHealthFit().getWeight(last3Days, now);
```

File: example/lib/main.dart [hunk 6]
```diff
@@ -27,14 +28,26 @@ class _MyAppState extends State<MyApp> {
...
+    setState(() {
+      _lastWeightString = lastWeight.toString();
```
Comment (avi-mastov) [inline example/lib/main.dart:36]: Didn't handle `lastWeight == null`.

File: android/src/main/kotlin/com/metaflow/flutterhealthfit/FlutterHealthFitPlugin.kt (hunk 1 above)
Comment (elnatantikal) [reply android/src/main/kotlin/com/metaflow/flutterhealthfit/FlutterHealthFitPlugin.kt:135]: fixed

**Pull request merged**